js_modules/dagit/packages/core/src/workspace/VirtualizedGraphTable.tsx

```diff
@@ -0,0 +1,144 @@
+import {gql, useLazyQuery} from '@apollo/client';
+import {Box, Caption, Colors} from '@dagster-io/ui';
+import {useVirtualizer} from '@tanstack/react-virtual';
+import * as React from 'react';
+import {Link} from 'react-router-dom';
+import styled from 'styled-components/macro';
+
+import {Container, HeaderCell, Inner, Row, RowCell} from '../ui/VirtualizedTable';
+
+import {useDelayedRowQuery} from './VirtualizedWorkspaceTable';
+import {RepoAddress} from './types';
+import {SingleGraphQuery, SingleGraphQueryVariables} from './types/SingleGraphQuery';
+import {workspacePathFromAddress} from './workspacePath';
+
+export type Graph = {name: string; path: string; description: string | null};
+
+interface Props {
+  graphs: Graph[];
+  repoAddress: RepoAddress;
+}
+
+export const VirtualizedGraphTable: React.FC<Props> = ({repoAddress, graphs}) => {
+  const parentRef = React.useRef<HTMLDivElement | null>(null);
+
+  const rowVirtualizer = useVirtualizer({
+    count: graphs.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 64,
+    overscan: 10,
+  });
+
+  const totalHeight = rowVirtualizer.getTotalSize();
+  const items = rowVirtualizer.getVirtualItems();
+
+  return (
+    <>
+      <Box
+        border={{side: 'horizontal', width: 1, color: Colors.KeylineGray}}
+        style={{
+          display: 'grid',
+          gridTemplateColumns: '100%',
+          height: '32px',
+          fontSize: '12px',
+          color: Colors.Gray600,
+        }}
+      >
+        <HeaderCell>Graph</HeaderCell>
+      </Box>
+      <div style={{overflow: 'hidden'}}>
+        <Container ref={parentRef}>
+          <Inner $totalHeight={totalHeight}>
+            {items.map(({index, key, size, start}) => {
+              const row: Graph = graphs[index];
+              return (
+                <GraphRow
+                  key={key}
+                  name={row.name}
+                  description={row.description}
+                  path={row.path}
+                  repoAddress={repoAddress}
+                  height={size}
+                  start={start}
+                />
+              );
+            })}
+          </Inner>
+        </Container>
+      </div>
+    </>
+  );
+};
+
+interface GraphRowProps {
+  name: string;
+  path: string;
+  description: string | null;
+  repoAddress: RepoAddress;
+  height: number;
+  start: number;
+}
+
+const GraphRow = (props: GraphRowProps) => {
+  const {name, path, description, repoAddress, start, height} = props;
+
+  const [queryGraph, queryResult] = useLazyQuery<SingleGraphQuery, SingleGraphQueryVariables>(
+    SINGLE_GRAPH_QUERY,
+    {
+      fetchPolicy: 'cache-and-network',
+      variables: {
+        selector: {
+          repositoryName: repoAddress.name,
+          repositoryLocationName: repoAddress.location,
+          graphName: name,
+        },
+      },
+    },
+  );
+
+  useDelayedRowQuery(queryGraph);
+  const {data} = queryResult;
+
```

Comment:
I don't know that the lazy per-row query is super worth it for just the descriptions - as far as I've seen they're not usually present.  I guess it's good to have this stuff in place for future use, though.

Reply:
Yeah it felt a little weird on this since there's nothing else there. But I figured there might be more we want to add that we've never bothered to add for Graphs, so I put it here. 🤷